### DISCUSSION TITLE
Programming C++ and Comiling it to .bin

### DISCUSSION THREAD

**Discussion by BalrajArpit** [eclipse-theia/theia#8793] [Q&A]
I want to compile C++ codes into .bin so that anyone could write, build and share binary fill to others to execute.

**Comment by vince-fugnitto** [comment]
@BalrajArpit thank you for your discussion, could you please clarify what your specific question is and include additional information?

**Reply by BalrajArpit** [reply]
I want to use Theia to compile code for ESP8266 and ESP32 in myfile.bin format and save it to a folder as firmware/myfile.bin for OTA update.

Reference: https://arduino-esp8266.readthedocs.io/en/latest/ota_updates/readme.html#http-server

**Reply by vince-fugnitto** [reply]
@BalrajArpit I'm not familiar with the information you referenced, but it would seem as though you'd require custom implementation to support the feature in either a `@theia` extension, or a `vscode` extension consumed by your custom `theia`-based application. If you cannot already find extensions in the [open-vsx](https://open-vsx.org/) marketplace that can be used to satisfy your requirements, a new custom implementation would most certainly be needed. 

Please note that there is the [arduino pro ide](https://www.arduino.cc/pro/arduino-pro-ide) (which is based on `theia`), which may already satisfy your requirements.
- https://github.com/arduino/arduino-pro-ide

**Reply by BalrajArpit** [reply]
Arduino Pro IDE is offline, I want to build online IDE. Can you refer me to any documentation where I can integrate SDK (https://www.espressif.com/en/support/download/sdks-demos) into Theia to compile programs?

**Reply by vince-fugnitto** [reply]
> Arduino Pro IDE is offline, I want to build online IDE. Can you refer me to any documentation where I can integrate SDK (https://www.espressif.com/en/support/download/sdks-demos) into Theia to compile programs?

@BalrajArpit unfortunately, there is no such documentation of advanced use-cases to integrate `sdks` into a custom application.
You must create your own [theia-extension](https://theia-ide.org/docs/authoring_extensions/) to implement/integrate the `sdk` yourself, extending the base framework.

**Reply by kittaakos** [reply]
@BalrajArpit, it's not clear what you want to achieve 👇 

> I want to compile C++ codes into .bin

> I want to build online IDE

but if you want to compile for ESP8266 board in an online environment, you might want to take a look into [Arduino Create](https://www.arduino.cc/en/Main/Create/).

**Reply by sakthivel-axim** [reply]
@BalrajArpit Did you get any solution for your requirement ?
